pyproject.toml

```diff
@@ -71,11 +80,11 @@ dependencies = [
     "ruff~=0.3.4",
     "types-PyYAML~=6.0.12",
     "types-requests~=2.31.0",
-    "databricks-connect~=15.4",
-    "pyspark~=3.5.0",
+    "databricks-connect~=16.4.3",
```

Comment:
i think it's not possible to use newer version of databricks connect on older runtimes. So this won't work on our tests cluster that has 15.4. We require runtime 15.4 or higher for dqx so we should still support the older one.